### DISCUSSION TITLE
Text or Tabular predictor for  sentence similarity?

### DISCUSSION THREAD

**Discussion by Wang-Lin-boop** [autogluon/autogluon#1703] [Q&A] [answered]
For the sentence similarity task, how do I decide whether to use a Text predictor or a Tabular predictor?

It is worth mentioning that neither ```leaderboard()``` nor ```fit_summary(show_plot=True) ``` mentioned in the cheat sheet can be used in the text predictor. In addition, the API and options for Tabular predictors are clearly richer than those for Text predictors.

So I would like to know if the Text predictor has better performance on text tasks compared to the Tabular predictor?

**Comment by Innixma** [comment, marked as answer]
You can try both, or use both at the same time through Tabular via `hyperparameters='multimodal'` (TextPredictor is used as a model in TabularPredictor's ensemble). You can even have images in your dataset along with text and tabular and it will all work seamlessly using TabularPredictor!

Here is the tutorial: https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-multimodal.html

Combo this with `presets='best_quality'` or `presets='high_quality'` and you will have a (albeit expensive to train) extremely powerful solution.
